src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -53,17 +65,24 @@ def invalid_metric_tags(tags: Mapping[str, str]) -> Sequence[str]:
     return invalid_strs
 
 
+class InboundMessage(TypedDict):
+    # Note: This is only the subset of fields we access in this file.
+    org_id: int
+    name: str
+    type: str
+    tags: Mapping[str, str]
+
+
 class IndexerBatch:
```

Comment:
Please add a unit test for the logic you added to this class.
- set up a batch with messages
- filter it out with `limit_messages` 
- reconstruct messages and see the filtered one are skipped.
- extract strings and ensure the strings in the messages you filtered are not there.
(also let's see that if a string is present both in a message that is filtered out and in a message that is returned, the string is returned)

Reply:
done